Backpack-SwiftUI/FlightLeg/Classes/BPKFlightLeg.swift

```diff
@@ -0,0 +1,177 @@
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        if let carrierLogo = carrierLogo {
+            self._carrierLogo = State(initialValue: Image(uiImage: carrierLogo))
+        } else {
+            self._carrierLogo = State(initialValue: nil)
+        }
+    }
+
+    var body: some View {
+        BPKDynamicStack(horizontalAlignment: .leading, verticalAlignment: .top, spacing: .base) {
+            createCarrierLogo()
+            createFlightInfo()
+                .frame(
+                    maxWidth: .infinity,
+                    alignment: .leading
+                )
+            createDurationInfo()
+        }
+    }
+
+    @ViewBuilder
+    private func createCarrierLogo() -> some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            if let carrierLogo = carrierLogo {
+                carrierLogo
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: airlineLogoSize, height: airlineLogoSize)
+                    .background(BPKColor.textOnDarkColor)
+                    .cornerRadius(BPKCornerRadius.xs.value)
+            }
+        }
+        .padding(.top, .sm)
+    }
+
+    @ViewBuilder
+    private func createFlightInfo() -> some View {
+        VStack(alignment: .leading, spacing: 0) {
+            Text(departureArrivalTimeAttributedText() + nextDayAttributedText())
+                .foregroundColor(.textPrimaryColor)
+                .lineLimit(nil)
+                .accessibilityElement(children: .combine)
+
+            BPKText(flightDescription, style: .caption)
+                .foregroundColor(.textSecondaryColor)
+                .lineLimit(nil)
+
+            if let operatedBy = operatedBy {
+                BPKText(operatedBy, style: .caption)
+                    .foregroundColor(.textSecondaryColor)
+                    .lineLimit(nil)
+            }
+
+            if let warning = warning {
+                BPKText(warning, style: .label3)
+                    .foregroundColor(.textErrorColor)
+                    .lineLimit(nil)
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func createDurationInfo() -> some View {
+        VStack(alignment: .trailing, spacing: .sm) {
+            BPKText(stopsInfo, style: .label3)
+                .foregroundColor(highlightStopsInfo ? .textErrorColor : .textPrimaryColor)
+
+            BPKText(duration, style: .caption)
+                .foregroundColor(.textSecondaryColor)
+        }
+    }
+
+    private func departureArrivalTimeAttributedText() -> AttributedString {
+        var str = AttributedString(departureArrivalTime)
+        str.font = BPKFontStyle.heading5.font
+
+        return str
+    }
+
+    private func nextDayAttributedText() -> AttributedString {
+        var str = AttributedString(nextDayArrival ?? "")
```

Comment:
why create this attributedString if the nextDayArrival is empty/nil? should this not return an optional attributedString?